vitrina/projects/services.py

```diff
@@ -1,46 +1,94 @@
 from vitrina.users.models import User
 from vitrina.projects.models import Project
-from vitrina.orgs.services import has_perm, is_representative, Action
-from vitrina.orgs.models import Representative, Organization
 from django.db.models import Q
-from django.contrib.contenttypes.models import ContentType
 from django.db.models import QuerySet
+from django.contrib.auth.models import AnonymousUser
 
 
-def can_update_project(user: User, project: Project) -> bool:
+def _q_visible_projects(user: User | AnonymousUser) -> Q:
+    public_approved = Q(is_public=True, status=Project.APPROVED)
+
+    if not getattr(user, "is_authenticated", False):
+        return public_approved
+
     if user.is_staff or user.is_superuser:
-        return True
+        return Q()
+
+    represented_org_ids = user.represented_org_ids
+
+    filter = (
+        public_approved
+        # Owner can view personal projects
+        | Q(organization__isnull=True, user=user)
+        # All organization representatives can view organization's projects
+        | Q(organization_id__in=represented_org_ids)
+        # Representatives can view projects of organizations, they represent
+        | Q(datasets__organization_id__in=represented_org_ids)
+    )
```

Comment:
Optional.
According to some of good practices your comments should answer to question `Why` and not `What`. If you want to make more clear `What` this code does here - user variable and methods names, like:

```suggestion

        is_owner = Q(organization__isnull=True, user=user)
        is_org_representetive = Q(organization_id__in=represented_org_ids)
        is_dataset_representetive = Q(datasets__organization_id__in=represented_org_ids)
        
    filter = public_approved | is_owner  | is_org_representetive | is_dataset_representetive
```